stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamConfiguration.kt

```diff
@@ -0,0 +1,27 @@
+package io.getstream.chat.android.compose.ui.theme
+
+import android.content.Context
+import com.getstream.sdk.chat.utils.DateFormatter
+
+/**
+ * Central place for general app configuration options. Use this class to define how some of the features in the app
+ * work.
+ *
+ * @param dateFormatter - The formatter used for timestamps across the app.
+ * @param enforceUniqueReactions - Flag that enables or disables unique reactions on messages.
+ * */
+public class StreamConfiguration(
+    public val dateFormatter: DateFormatter,
+    public val enforceUniqueReactions: Boolean,
```

Comment:
Let's consider what exactly should go here vs into separate parameters in ChatTheme. Would be nice to set guidelines for this somewhere.

Reply:
Do you think we can move the `reactionOptions` here?

E.g. the guidelines can be something like:
- Anything that's a theming option/direct UI impact has its own provider
- Anything that's state-based, business logic affecting goes here?

We can discuss more if this makese sense in the long run (e.g. if something is both UI & logic/state based)

Reply:
Not an easy question. For example, both `reactionTypes` and `attachmentFactories` have UI-related code directly in them, so according to those guidelines, they should not be inside the config. Maybe `dateFormatter` shouldn't be either 😅